dkvk-ios/DKVK/Classes/Screens/CreatePost/CreatePostViewController.swift

```diff
@@ -8,39 +8,93 @@
+    @IBOutlet private weak var postImageView: UIImageView!
+    @IBOutlet private weak var crossImageView: UIImageView!
+
+    private let tapToAddPostImage = #imageLiteral(resourceName: "tap_button")
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        Decorator.decorate(vc: self)
+
+        addTargets()
+        addGestures()
+
+        updatePostImageView(image: tapToAddPostImage)
+    }
+}
+
+private extension CreatePostViewController {
+
+    func addTargets() {
+        doneButton.addTarget(self, action: #selector(doneButtonClicked), for: .touchUpInside)
+    }
+
+    func addGestures() {
+        let postImageTapGesture = UITapGestureRecognizer(target: self, action: #selector(postImageClicked))
+        postImageView.addGestureRecognizer(postImageTapGesture)
+
+        let crossTapGesture = UITapGestureRecognizer(target: self, action: #selector(crossImageClicked))
+        crossImageView.addGestureRecognizer(crossTapGesture)
+    }
+
+    @objc func postImageClicked() {
+        let imagePickerController = UIImagePickerController()
+        imagePickerController.delegate = self
+        imagePickerController.sourceType = .photoLibrary
+        present(imagePickerController, animated: true, completion: nil)
+    }
+
+    @objc func crossImageClicked() {
+        updatePostImageView(image: tapToAddPostImage)
+    }
+
+    @objc func doneButtonClicked() {
+
+        guard let user = AuthManager.shared.currentUser else {
+            showAlert(with: "Error", and: "User not logged in")
+            return
+        }
+
+        let postImage = postImageView.image != #imageLiteral(resourceName: "tap_button.png") ? postImageView.image : nil
+
+        PostManager.shared.createPost(from: user, with: textView.text, image: postImage) { (result) in
+            switch result {
+            case .error(let textError):
+                self.showAlert(with: "Error", and: textError)
+            case .success:
+                self.showAlert(with: "Success", and: "Post has been created")
+            }
+        }
+    }
+
+    func updatePostImageView(image: UIImage) {
+        postImageView.image = image
+        crossImageView.isHidden = image == tapToAddPostImage
+    }
+}
+
+private extension CreatePostViewController {
+    final class Decorator {
+        static func decorate(vc: CreatePostViewController) {
+            vc.postImageView.isUserInteractionEnabled = true
+            vc.crossImageView.isUserInteractionEnabled = true
+            vc.textView.placeholder = "Add post message..."
+        }
+    }
+}
+
+extension CreatePostViewController: UINavigationControllerDelegate, UIImagePickerControllerDelegate {
+    func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [UIImagePickerController.InfoKey : Any]) {
+        picker.dismiss(animated: true, completion: nil)
+        guard let image = info[UIImagePickerController.InfoKey.originalImage] as? UIImage else {
```

Comment:
need to use .current and if there is no current - get original

Reply:
Sorry. `.original` is for Photos framework. There is `. editedImage `
https://developer.apple.com/documentation/uikit/uiimagepickercontroller/infokey/1619134-editedimage

Reply:
It is very interesting.
I have checked it now.
Yeap. it nil for `.photoLibrary` and `.isEditing = false`.
I thought it working like Photos framework but i was wrong.
Thank you!

Any way. We can keep new logic if we will need camera + edit mode :-)

Reply:
by the way can be written `info[.editedImage]`

Reply:
If add `imagePickerController.allowsEditing = true` then info[.editedImage] will be available

Reply:
> If add `imagePickerController.allowsEditing = true` then `info[.editedImage]` will be available

As I understood, it is right only for camera type, not for photoLibrary.